app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -56,6 +60,27 @@ open class FenixApplication : LocaleAwareApplication() {
             return
         }
 
+        // We need to always initialize Glean and do it early here. Note that we are disabling it
+        // here too (uploadEnabled = false). If needed Glean will be enabled later by the migration
+        // code (if this user used to be a fennec user with the right flags enabled) or by
+        // GleanMetricsService if telemetry is enabled for this user.
+        // It is important that this initialization happens *here* before calling into
+        // setupInMainProcessOnly() which behaves differently for fenix and fennec builds.
+        // Glean needs to be disabled initially because otherwise we may already collect telemetry
+        // before we know whether we want that (which is *after* the migration).
+        // As a side effect this means pings submitted between the initialization here and until we
+        // potentially enable Glean would be lost. However such pings do not exist at this moment.
```

Comment:
GV may emit something, but that's done after our blocking/telemetry migration, so it should be fine.